src/Uno.UI/UI/Xaml/Automation/Peers/FrameworkElementAutomationPeer.cs

```diff
@@ -1,34 +1,147 @@
-#pragma warning disable 108 // new keyword hiding
 using System;
 using Uno;
+using Uno.UI;
+using System.Linq;
+using Windows.UI.Xaml.Controls;
+
+#if __ANDROID__
+using View = Android.Views.ViewGroup;
+#elif __IOS__
+using View  = UIKit.UIView;
+using UIKit;
+#else
+using View  = Windows.UI.Xaml.UIElement;
+#endif
 
 namespace Windows.UI.Xaml.Automation.Peers
 {
-	public  partial class FrameworkElementAutomationPeer : global::Windows.UI.Xaml.Automation.Peers.AutomationPeer
+	public partial class FrameworkElementAutomationPeer : AutomationPeer
 	{
-		[NotImplemented]
+		public IFrameworkElement Owner { get; } // TODO: UIElement
+
+		public FrameworkElementAutomationPeer() { }
+
+		public FrameworkElementAutomationPeer(object element)
+		{
+			Owner = element as IFrameworkElement; // TODO: UIElement
+		}
+
 		public FrameworkElementAutomationPeer(IFrameworkElement element)
 		{
-			throw new NotImplementedException();
+			Owner = element;
 		}
 
-		[NotImplemented]
-		public FrameworkElementAutomationPeer(object element)
+		public FrameworkElementAutomationPeer(FrameworkElement owner) : base()
+		{
+			Owner = owner;
+		}
+
+		public static AutomationPeer CreatePeerForElement(IFrameworkElement element) // TODO: UIElement
+		{
+			if (element == null)
+			{
+				throw new ArgumentNullException(nameof(element));
+			}
+
+			return element.GetAutomationPeer();
+		}
+
+		public static AutomationPeer FromElement(IFrameworkElement element) // TODO: UIElement
+		{
+			if (element == null)
+			{
+				throw new ArgumentNullException(nameof(element));
+			}
+
+			return element.GetAutomationPeer();
+		}
+
+		protected override string GetLocalizedControlTypeCore()
+		{
+			if (AutomationProperties.GetLocalizedControlType(Owner) is string localizedControlType && !string.IsNullOrEmpty(localizedControlType))
+			{
+				return localizedControlType;
+			}
+
+			return base.GetLocalizedControlTypeCore();
+		}
+
+		protected override AutomationPeer GetLabeledByCore()
+		{
+			if (AutomationProperties.GetLabeledBy(Owner) is IFrameworkElement label) // TODO: UIElement
+			{
+				return label.GetAutomationPeer();
+			}
+
+			return base.GetLabeledByCore();
+		}
+
+		protected override string GetNameCore()
 		{
-			throw new NotImplementedException();
+			if (AutomationProperties.GetName(Owner) is string name && !string.IsNullOrEmpty(name)) // TODO: Empty check?
+			{
+				return name;
+			}
+
+			if (GetLabeledBy() is AutomationPeer labelAutomationPeer && labelAutomationPeer.GetName() is string label && !string.IsNullOrEmpty(label))
+			{
+				return label;
+			}
+
+			if (FeatureConfiguration.AutomationPeer.UseSimpleAccessibility)
+			{
+				if (Owner is View view && AutomationProperties.GetAccessibilityView(Owner) != AccessibilityView.Raw)
+				{
+					// TODO: How expensive is this? Is this called recursively for all children?
```

Comment:
Is it?

Reply:
I refactored that part, confirmed this wasn't a problem, and explained why.